src/components/AttachmentModal.js

```diff
@@ -74,22 +76,34 @@ class AttachmentModal extends PureComponent {
 
         this.state = {
             isModalOpen: false,
+            isConfirmModalOpen: false,
+            reportId: null,
+            showArrows: false,
+            file: {name: lodashGet(props, 'originalFileName', '')},
             isAttachmentInvalid: false,
             attachmentInvalidReasonTitle: null,
             attachmentInvalidReason: null,
-            file: null,
-            sourceURL: props.sourceURL,
             modalType: CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE,
             isConfirmButtonDisabled: false,
             confirmButtonFadeAnimation: new Animated.Value(1),
+            sourceURL: props.isAuthTokenRequired ? addEncryptedAuthTokenToURL(props.sourceURL) : props.sourceURL,
         };
 
         this.submitAndClose = this.submitAndClose.bind(this);
         this.closeConfirmModal = this.closeConfirmModal.bind(this);
+        this.onNavigate = this.onNavigate.bind(this);
         this.validateAndDisplayFileToUpload = this.validateAndDisplayFileToUpload.bind(this);
         this.updateConfirmButtonVisibility = this.updateConfirmButtonVisibility.bind(this);
     }
 
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} param
+     */
+    onNavigate(param) {
+        this.setState(param);
+    }
```

Comment:
What param contains can you explain?

Reply:
May I thing a comment could be useful  here

Reply:
this is the data for attachments I was thinking 
```suggestion
     * Helps to navigate between next/previous attachments
     * by setting sourceURL and file in state
     * @param {Object} attachmentData
     */
    onNavigate(attachmentData) {
        this.setState(attachmentData);
    }
```